passes/techmap/iopadmap.cc

```diff
@@ -415,15 +334,47 @@ struct IopadmapPass : public Pass {
 				{
 					RTLIL::Cell *cell = module->addCell(NEW_ID, RTLIL::escape_id(celltype));
 					cell->setPort(RTLIL::escape_id(portname), RTLIL::SigSpec(wire));
-					if (!portname2.empty())
+
+					if (!portname2.empty()) {
+						RTLIL::Wire *new_wire = NULL;
+						new_wire = module->addWire(NEW_ID, wire);
+						module->swap_names(new_wire, wire);
+						wire->attributes.clear();
 						cell->setPort(RTLIL::escape_id(portname2), RTLIL::SigSpec(new_wire));
+					}
 					if (!widthparam.empty())
 						cell->parameters[RTLIL::escape_id(widthparam)] = RTLIL::Const(wire->width);
 					if (!nameparam.empty())
 						cell->parameters[RTLIL::escape_id(nameparam)] = RTLIL::Const(RTLIL::id2cstr(wire->name));
 					cell->attributes[ID::keep] = RTLIL::Const(1);
 				}
 
+				if (!rewrite_bits.count(wire)) {
+					wire->port_id = 0;
+					wire->port_input = false;
+					wire->port_output = false;
+				}
+			}
+
+			for (auto &it : rewrite_bits) {
+				RTLIL::Wire *wire = it.first;
+				RTLIL::Wire *new_wire = module->addWire(NEW_ID, wire);
+				module->swap_names(new_wire, wire);
+				wire->attributes.clear();
```

Comment:
Is it desirable to remove all attributes on this wire? What if there was a user attribute?

Reply:
Then the user attributes will now be on the new wire.  This dance is a roundabout way of creating a new plain anonymous wire and reconnecting all cells/connects from the original wire to it — since it turns out to be simpler to move the name and attributes to a new wire than to move connections, we do it this way.

Reply:
Pardon me if I'm missing something, but all I see is `wire->attributes.clear()` and I'm not seeing where `new_wire` adopts those attributes...?

Reply:
Right when it is created — `module->addWire(NEW_ID, wire)` copies the attributes of the original wire.

Reply:
Ah-ha! Thanks for pointing that out.